tests/login.spec.ts

```diff
@@ -0,0 +1,223 @@
+            await expect(await loginPage.getProfileDropdownMenu()).toBeVisible();
+            await expect(await loginPage.getProfileDropdownMenuEmail()).toBeVisible();
+            await expect(await loginPage.getProfileDropdownMenuEmail()).toHaveText(user.email || "undefined");
+
+            await loginPage.clickLogout();
+            await expect(await loginPage.getEnterButton()).toBeVisible();
+        }
+    });
+
+    test("Authorization with valid phone and password", async ({ page }) => {
+        const phones = [
+            process.env.USER_PHONE,
+            process.env.USER_PHONE?.substring(1),
+            process.env.USER_PHONE?.substring(3)
+        ];
+
+        for (const phone of phones) {
+            await loginPage.clickEnterButton();
+            await loginPage.fillEmailInput(phone);
+            await expect(await loginPage.getEmailInput()).toHaveValue(phone || "undefined");
+            await loginPage.emailInputIsNotHighlighted();
+
+            await loginPage.fillPasswordInput(process.env.USER_PASSWORD);
+            await expect(await loginPage.getPasswordInput()).toHaveValue(process.env.USER_PASSWORD || "undefined");
+            await loginPage.passwordInputIsNotHighlighted();
+
+            await loginPage.clickLoginButton();
+            await loginPage.waitForAuthorizationFormIsHidden();
+            await mainPage.isOpen();
+
+            await expect(await loginPage.getProfileIcon()).toBeVisible();
+            await loginPage.clickProfileIcon();
+            await expect(await loginPage.getProfileDropdownMenu()).toBeVisible();
+
+            await loginPage.clickMyProfile();
+            await page.waitForURL("/owner-cabinet/", { waitUntil: "networkidle" });
+            await expect(await profilePage.getPhoneNumberInput()).toBeVisible();
+            await expect(await profilePage.getPhoneNumberValue()).toContain(phone || "undefined");
+            await expect(await profilePage.getPhoneVerificationLabel()).toBeVisible();
+            await expect(await profilePage.getPhoneVerificationLabel()).toHaveText("Успішно верифіковано");
+
+            await profilePage.clickLogout();
+            await mainPage.isOpen();
+            await expect(await loginPage.getEnterButton()).toBeVisible();
+        }
+    });
+
+    test("Authorization with invalid phone", async ({ page }) => {
+        await loginPage.clickEnterButton();
+
+        await loginPage.fillPasswordInput(process.env.USER_PASSWORD);
+        await expect(await loginPage.getPasswordInput()).toHaveValue(process.env.USER_PASSWORD || "undefined");
+        await loginPage.passwordInputIsNotHighlighted();
+
+        const invalidPhones = [
+            "991234785",
+            "099123478",
+            "+380-99-123-4785",
+            "+380 99 123 4785",
+            "+380(99)1234785",
+            "(99)1234785",
+            "09912347850",
+            "+100991234785",
+            "+0991234785"
+        ];
```

Comment:
Here and further, instead of using hardcoded data, you can use some randomizer to avoid the Pesticide Paradox loop.

Reply:
The hardcoded data are designed to represent specific patterns and edge cases (e.g., missing digits, incorrect country codes, extra characters) for the specific account. When dealing with invalid cases, it's crucial to know exactly what format caused the validation to fail. Randomized data might lead to inconsistent results or missed validations over time.

Reply:
You can configure the randomizer function to pull the data construction you want, just to avoid repeating. This may be a non-trivial way, but it's definitely the best.

Reply:
It can work for phone numbers, but how about the email and the password? They all are mostly the same, only incorrect symbols are added to them